server/server_test.go

```diff
@@ -223,3 +226,25 @@ func TestNoDeadlockOnStartFailure(t *testing.T) {
 	// We should be able to shutdown
 	s.Shutdown()
 }
+
+func TestMaxConnections(t *testing.T) {
+	opts := DefaultOptions
+	opts.MaxConn = 1
+	s := RunServer(&opts)
+	defer s.Shutdown()
+
+	addr := fmt.Sprintf("nats://%s:%d", opts.Host, opts.Port)
+	nc, err := nats.Connect(addr)
+	if err != nil {
+		t.Fatalf("Error creating client: %v\n", err)
+	}
+	defer nc.Close()
+
+	nc2, err := nats.Connect(addr)
+	if err == nil {
+		if nc2 != nil {
+			nc2.Close()
```

Comment:
No we can't, but I should not check for `nc2 != nil` since if `err == nil`, it means that `nc2` is not nil!